gapic-generator/templates/default/service/client/method/def/_request_normal.erb

```diff
@@ -1,14 +1,17 @@
 <%- assert_locals method -%>
 raise ::ArgumentError, "request must be provided" if request.nil?
-<%- if method.auto_populated_fields.length > 0 -%>
-
-# Auto populate fields for this request.
-<%- end -%>
 <% method.auto_populated_fields.each do |field| %>
+<% name = field.name %>
+
+# Auto populate request field `<%= name %>`.
 if request.is_a? Hash
-  request[:<%= field %>] = SecureRandom.uuid unless request.key? :<%= field %>
+  request[:<%= name %>] = SecureRandom.uuid unless request.key? :<%= name %>
```

Comment:
Now that I look at this again, can you also check for the string version of the key, in addition to the symbol version? Using strings as keys is not common, but supported. Same with the streaming version below.

Reply:
Thanks! I added the check and two more test cases (for hash with symbol vs string). Plus a minor bug fix with `SecureRandon.define_singleton_method` that I found impacting other tests downstream.

Please feel free to add more comments before I merge. Otherwise, I will do submit it Monday.